tasks/gather_facts_subnets.yml

```diff
@@ -75,3 +75,5 @@
     - "{{ aws_vpc_rt_subnet_by_cidr }}"
     - "{{ aws_vpc_rt_subnet_by_filter }}"
     - "{{ aws_vpc_rt_subnet_by_id }}"
+  when: not ([item] in aws_vpc_route_table_subnets)
```

Comment:
I'd say we should assert here and error out in order to make the *user* aware of the fact that two duplicate subnets have been found.

Reply:
Is that really an error? Could we not in theory have two subnets with duplicated facts but in different fields? If it is, then I would agree, but it sounds to me that it might not be in all cases.

Reply:
In particular, maybe two subnets with the same tags?

Reply:
Oops, wait a second. This tasks cannot gather subnets by tags!

Reply:
It is search by filter. So you could for example only specify `tag:department=DevOps` which would return a lot of subnets which all have this tag attached.

Reply:
Well, then the question is whether or not we would have some sort of logic to detect when filters are given which return too many subnets? Perhaps a ticket should be opened to deal with this later on? For now the changed I made will keep duplicate subnets from being added to the route table subnet list.